File: Chapter-3/Java/Problem9.java
import java.util.Scanner;

public class Problem9 {
    public static boolean is_leap(int year) {
        if (year % 4 == 0 && year % 100 != 0 || year % 400 == 0) {
            return true;
        } else {
            return false;
        }
    }

    public static void main(String args[]) {
        int year, odd_days = 0;
        Scanner sc = new Scanner(System.in);
        year = sc.nextInt();
        for (int i = 0; i < year; i++) {
            if (is_leap(year)) {
                odd_days += 366;
            } else {
                odd_days += 365;
            }
        }
        odd_days %= 7;
        String[] weekdays = { "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday" };
        System.out.printf("The day of the week at 01/01/%d is %s", year, weekdays[odd_days]);
        sc.close();
    }
}

File: Chapter-3/Python/Problem9.py
def is_leap(year):
    if year % 4 == 0 and year % 100 != 0 or year % 400 == 0:
        return True
    else:
        return False

if __name__ == '__main__':
    odd_days = 0
    year = int(input("Enter a year: "))
    for i in range(1,year):
        if is_leap(year):
            odd_days += 366
        else:
            odd_days += 365
    
    odd_days %= 7

    weekdays = ["Monday","Tuesday","Wednesday","Thursday","Friday","Saturday","Sunday"]
    print(f"The day of the week at 01/01/{year} is {weekdays[odd_days]}")

